### PR TITLE
Make `unsizing_params_for_adt` into a query

### PR DESCRIPTION
**Pull request by compiler-errors**
Addresses a FIXME in confirmation.

r? @ghost

### CODE REVIEW

**Comment by compiler-errors** [comment]
I'm inclined to land this even if it's perf neutral, since it'll help us share some unsizing code between the old and new trait solvers.

r? compiler

File: compiler/rustc_middle/src/query/mod.rs [hunk 1]
```diff
@@ -183,6 +183,14 @@ rustc_queries! {
         separate_provide_extern
     }
 
+    query unsizing_params_for_adt(key: DefId) -> &'tcx rustc_index::bit_set::BitSet<u32>
+    {
+        desc { |tcx|
```
Comment (cjgillot) [inline compiler/rustc_middle/src/query/mod.rs:188]:
```suggestion
    query unsizing_params_for_adt(key: DefId) -> rustc_index::bit_set::BitSet<u32> {
        arena_cache
        desc { |tcx|
```
And we can let the query system do the arena allocation on its own.

**Review by cjgillot — COMMENTED** [review]
r=me with one nit

**Comment by compiler-errors** [comment]
This isn't perf sensitive

@bors rollup-

**Pull request merged**